### DISCUSSION TITLE
error mesage while remapping

### DISCUSSION THREAD

**Discussion by Senne98** [Alvinn8/paper-nms-maven-plugin#7] [Q&A]
Can you help me whit the remapping, I asked somwhere else and they told me I had to ask the auther.
`Cannot invoke "java.io.File.toPath()" because the return value of "org.apache.maven.artifact.Artifact.getFile()" is null`
and this is my pom.xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>

    <groupId>com.Super_man2006.Shop</groupId>
    <artifactId>Skyblock</artifactId>
    <version>1.0-SNAPSHOT</version>
    <packaging>jar</packaging>

    <name>Skyblock</name>

    <description>disc</description>
    <properties>
        <java.version>17</java.version>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    </properties>

    <pluginRepositories>
        <pluginRepository>
            <id>bytecode.space</id>
            <url>https://repo.bytecode.space/repository/maven-public/</url>
        </pluginRepository>
    </pluginRepositories>

    <build>
        <directory>C:\Users\Senne\Desktop\plugins\server\plugins</directory>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.8.1</version>
                <configuration>
                    <source>${java.version}</source>
                    <target>${java.version}</target>
                </configuration>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-shade-plugin</artifactId>
                <version>3.2.4</version>
                <executions>
                    <execution>
                        <phase>package</phase>
                        <goals>
                            <goal>shade</goal>
                        </goals>
                        <configuration>
                            <createDependencyReducedPom>false</createDependencyReducedPom>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
            <plugin>
                <groupId>ca.bkaw</groupId>
                <artifactId>paper-nms-maven-plugin</artifactId>
                <version>1.2.1</version>
                <executions>
                    <execution>
                        <phase>process-classes</phase>
                        <goals>
                            <goal>remap</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
        </plugins>
        <resources>
            <resource>
                <targetPath>.</targetPath>
                <filtering>true</filtering>
                <directory>src/main/resources</directory>
            </resource>
        </resources>
    </build>

    <repositories>
        <repository>
            <id>papermc</id>
            <url>https://repo.papermc.io/repository/maven-public/</url>
        </repository>
        <repository>
            <id>sonatype</id>
            <url>https://oss.sonatype.org/content/groups/public/</url>
        </repository>
        <repository>
            <id>Multiverse-Core</id>
            <url>https://repo.onarandombox.com/content/repositories/multiverse/</url>
        </repository>
        <repository>
            <id>Multiverse-Inventories</id>
            <url>https://repo.onarandombox.com/content/repositories/multiverse/</url>
        </repository>
    </repositories>

    <dependencies>
        <dependency>
            <groupId>ca.bkaw</groupId>
            <artifactId>paper-nms</artifactId>
            <version>1.18.2-SNAPSHOT</version>
            <scope>provided</scope>
        </dependency>
        <dependency>
            <groupId>io.papermc.paper</groupId>
            <artifactId>paper-api</artifactId>
            <version>1.18.2-R0.1-SNAPSHOT</version>
            <scope>provided</scope>
        </dependency>
        <dependency>
            <groupId>com.onarandombox.multiversecore</groupId>
            <artifactId>Multiverse-Core</artifactId>
            <version>4.3.1</version>
            <type>jar</type>
            <scope>provided</scope>
        </dependency>
        <dependency>
            <groupId>com.onarandombox.multiverseinventories</groupId>
            <artifactId>Multiverse-Inventories</artifactId>
            <version>4.2.3</version>
            <type>jar</type>
            <scope>provided</scope>
        </dependency>
    </dependencies>

</project>

**Comment by Alvinn8** [comment]
Hello! I was not able to reproduce the issue. Could you try without setting the build directory? I also wouldn't recommend setting your build directory to a server's plugins directory, if that is what you are doing. Because if you run `mvn clean` you will delete your plugins folder, for example.

**Comment by Senne98** [comment]
It still doesn't work
Do you know what the error means, because i can't find it anywhere.

**Comment by Alvinn8** [comment]
I'm not really sure then. I can't see the stack trace to the error so I'm not really sure where it's coming from. But looking at the usages of Artifact#getFile() I would guess that the path of the project's artifact being null, or the path of a dependency being null, but maven should resolve those beforehand. So I'm not really sure.
